lint/base_linter/ruby_linter.py

```diff
@@ -25,30 +24,25 @@ class RubyLinter(linter.Linter):
     """
 
     @classmethod
-    def initialize(cls):
-        """Perform class-level initialization."""
-        super().initialize()
-
-        if cls.executable_path is not None:
-            return
-
-        if not callable(cls.cmd) and cls.cmd:
-            cls.executable_path = cls.lookup_executables(cls.cmd)
-        elif cls.executable:
-            cls.executable_path = cls.lookup_executables(cls.executable)
-
-        if not cls.executable_path:
-            cls.disabled = True
+    @lru_cache(maxsize=None)
+    def can_lint(cls, syntax):
```

Comment:
Why do you override this? I'm not seeing it right now.

Reply:
They implemented `initialize` which is called eager at Sublime start up. 

`can_lint` is generally lazy, called *once* on `cls` before first use. Bc of that it is still called before we have an instance, and a view. Usually, the default implementation works only with global available executables. 

So, if `can_lint` is relatively lax here (optimistic), you can implement `context_sensitive_executable_path`, which is called before every `lint`. (I just thought `ruby` has probably the same virtual env stuff like all the other languages.) Of course, if you do something before every lint, it should be fast.